### ISSUE TITLE
APCA constants - consistent naming

### ISSUE THREAD

**Issue by svgeesus** [w3c/wcag3#225]
(I would raise an issue on the constants in WCAG 3, but they haven't been added yet until https://github.com/w3c/silver/pull/570 is merged)

## The problem

The [_Current APCA Constants ( 0.0.98G - W3 last changed Feb 15, 2021 )_](https://github.com/Myndex/apca-w3#current-apca-constants--0098g---w3-last-changed-feb-15-2021-) lists constants, but gives no equations.

The [_APCA Base Math (sRGB)_](https://github.com/Myndex/apca-w3#apca-base-math-srgb) lists constants and equations, but the names used are not the same and the set of constants is not exactly equivalent.

When implementing, I mainly used the value of the constants to check equivalence (which kind of defeats the point of named constants) but clearly this fails if two constants have the same value and the mapping is not immediately obvious.

## Correspondences

The correspondences seem to be:

<table>
<tr>
    <th>Constant</th>
    <th>Value</th>
    <th>Base Math</th>
</tr>

<tr><td colspan="3">exponents</td></tr>
<tr><td>mainTRC</td><td>2.4</td><td><i>no name</i></td></tr>
<tr><td>normBG</td><td>0.56</td><td><i>no name</i></td></tr>
<tr><td>normTXT</td><td>0.57</td><td><i>no name</i></td></tr>
<tr><td>revTXT</td><td>0.62</td><td><i>no name</i></td></tr>
<tr><td>revBG</td><td>0.65</td><td><i>no name</i></td></tr>

<tr><td colspan="3">colorSpace</td></tr>
<tr><td>sRco</td><td>0.2126729</td><td><i>no name</i></td></tr>
<tr><td>sGco</td><td>0.7151522</td><td><i>no name</i></td></tr>
<tr><td>sBco</td><td>0.0721750</td><td><i>no name</i></td></tr>

<tr><td colspan="3">clamp</td></tr>
<tr><td>blkThrs</td><td>0.022</td><td>B<sub>thresh</sub></td></tr>
<tr><td>blkClmp</td><td>1.414</td><td>B<sub>exp</sub></td></tr>
<tr><td>loClip</td><td>0.1</td><td>P<sub>out</sub></td></tr>
<tr><td>deltaYmin</td><td>0.0005</td><td>P<sub>in</sub></td></tr>

<tr><td colspan="3">scalers</td></tr>
<tr><td>scaleBoW</td><td>1.14</td><td>R<sub>scale</sub></td></tr>
<tr><td>loBoWoffset</td><td>0.027</td><td>W<sub>offset</sub></td></tr>
<tr><td>scaleWoB</td><td>1.14</td><td>R<sub>scale</sub></td></tr>
<tr><td>loWoBoffset</td><td>0.027</td><td>W<sub>offset</sub></td></tr>

</table>

## Suggested fixes

1. Give the un-named constants names
2. Change B<sub>exp</sub> to B<sub>clmp</sub>, or otherwise make the correspondence obvious
3. Either change P<sub>out</sub> to something like Clip<sub>low</sub>, or change loClip to Pout
4. Either change P<sub>in</sub> to something like deltaY<sub>min</sub>, or change deltaYmin to Pin
5. Either decide that the WoB and BoW scalers and offsets might be different, and thus mint new names for use in base math (like BoW<sub>scale</sub>, WoB<sub>scale</sub>, BoW<sub>offset</sub>, WoB<sub>offset</sub> )or decide that they will be the same going forward and mint just two new names for the constants, like scale and offset.

## Also

In addition, please be aware that the chosen 5-digit chromaticity values for D65 differ slightly from the 4-digit ones used in CSS Color 4 (the latter being changed several times, although we finally went for the 4-digit ones throughout for consistency with the defining color space specifications) and also that the matrix coefficients for RGB to Y also differ from both CSS and the sRGB specification. The CSS ones were calculated from first principles and agree with the sRGB standard ones when rounded to 8 significant figures.

**Comment by Myndex** [comment]
> [!TIP]
> There is plain language documentation available at the following links

### [**_Easy Intro to APCA_**](https://git.apcacontrast.com/documentation/APCAeasyIntro)
An easy to read introduction to contrast for readability, with visual examples.

### [_Linktree of Essential Resources_](https://linktr.ee/Myndex)
A curated linktree of documentation, tools, and other resources for improving readability on the web.
